src/net_processing.cpp

```diff
@@ -3799,6 +3799,13 @@ bool PeerLogicValidation::SendMessages(CNode* pto, std::atomic<bool>& interruptM
                         connman->PushMessage(pto, msgMaker.Make(NetMsgType::INV, vInv));
                         vInv.clear();
                     }
+                    // Prepare an inv with a corresponding InstantSend lock to be sent later too
+                    uint256 islockHash;
+                    if (llmq::quorumInstantSendManager->GetInstantSendLockHashByTxid(hash, islockHash)) {
+                        LogPrint(BCLog::NET, "SendMessages -- preparing islock %s %s\n", hash.ToString(), islockHash.ToString());
+                        CInv islockInv(MSG_ISLOCK, islockHash);
+                        pto->PushInventory(islockInv);
```

Comment:
Any reason why you use PushInventory here and not `vInv` like in the code above? As I understand, the mempool request ignores if the requesting node already knows a TX and instead always sends INVs. We probably should do the same for IS lock invs.

Reply:
Good point 👍